### PR TITLE
plugins: address type errors for mypy uprev

### PR DESCRIPTION
**Pull request by cjp256**
Type fixes for plugins that should not affect runtime behavior.

### CODE REVIEW

File: snapcraft/plugins/cmake.py [hunk 1]
```diff
@@ -55,10 +55,11 @@ def __str__(self) -> str:
     def __init__(self, flag: str) -> None:
         parts = flag.split("=")
         self.name = parts[0]
+        self.value: Optional[str] = None
         try:
             self.value = parts[1]
         except IndexError:
-            self.value = None
+            pass
```
Comment (sergiusens) [inline snapcraft/plugins/cmake.py]: instead of passing, why not just do the assignment here and remove the assignment right before the try block?

Comment (cjp256) [reply snapcraft/plugins/cmake.py]: mypy complains that self.value is type string or whatever, and then it sees the except as trying to change it to optional.

but it probably would be better to do:

```
       try:
            self.value: Optional[str] = parts[1]
        except IndexError:
            self.value = None
```

Comment (cjp256) [reply snapcraft/plugins/cmake.py]: on a related note, would it be better form to check with len() rather than catch exception?

Comment (cjp256) [reply snapcraft/plugins/cmake.py]: Eh sorry, I fixed up commit and rebased against master at the same time.

```
diff --git a/snapcraft/plugins/cmake.py b/snapcraft/plugins/cmake.py
index 87d3a54a2..dbd826cb8 100644
--- a/snapcraft/plugins/cmake.py
+++ b/snapcraft/plugins/cmake.py
@@ -55,11 +55,10 @@ class _Flag:
     def __init__(self, flag: str) -> None:
         parts = flag.split("=")
         self.name = parts[0]
-        self.value: Optional[str] = None
         try:
-            self.value = parts[1]
+            self.value: Optional[str] = parts[1]
         except IndexError:
-            pass
+            self.value = None
```

Fixed!

**Pull request merged**